### ISSUE TITLE
Installation of MicroK8s 1.19 fails in Github Actions

### ISSUE THREAD

**Issue by knkski** [canonical/microk8s#2083]
See here for an example failure:

https://github.com/knkski/katib/runs/2058514778

```
echo install microk8s..
install microk8s..
exec sudo snap install microk8s --classic --channel=1.19/stable
microk8s (1.19/stable) v1.19.7 from Canonical* installed
echo creating microk8s group.
creating microk8s group.
exec sudo usermod -a -G microk8s $USER
echo creating default kubeconfig location.
exec mkdir -p '/home/runner/.kube/'
creating default kubeconfig location.
echo Generating kubeconfig file to default location.
Generating kubeconfig file to default location.
exec sudo microk8s kubectl config view --raw > $HOME/.kube/config
echo Change default location ownership.
exec sudo chown -f -R $USER $HOME/.kube/
Change default location ownership.
exec chmod go-rx $HOME/.kube/config
echo Start enabling dns
Start enabling dns
exec sudo microk8s enable dns
Enabling DNS
Applying manifest
serviceaccount/coredns created
configmap/coredns created
deployment.apps/coredns created
service/kube-dns created
clusterrole.rbac.authorization.k8s.io/coredns created
clusterrolebinding.rbac.authorization.k8s.io/coredns created
Restarting kubelet
error: error running snapctl: snap "microk8s" has "service-control" change in progress
Failed to enable dns
Error: exec: error: error running snapctl: snap "microk8s" has "service-control" change in progress

exec sudo microk8s status --wait-ready { silent: true }
exec sudo microk8s status --wait-ready { silent: true }
```

Switching the installed version to 1.20 made it work, see https://github.com/knkski/katib/actions/runs/633046913

**Comment by knkski** [comment]
It looks like these are the problem lines:

https://github.com/ubuntu/microk8s/blob/1.19/microk8s-resources/wrappers/apiservice-kicker#L39-L43

Running `git diff 1.18 1.19 microk8s-resources/wrappers/apiservice-kicker` shows that 1.19 switch from `systemctl restart ...` to `snapctl restart ...`, and the apiserver-kicker service is triggering the restarts in the background while we're enabling DNS. Not sure offhand what the difference between those two restart methods is that might be causing this.

Setting `--advertise-address=foo` in `$SNAP_DATA/args/kube-apiserver` should fix it in the short-term, but this seems like something that should get fixed in MicroK8s itself.

**Comment by ktsakalozos** [comment]
I was able to reproduce this (not reliably on 1.20 as well). What happens is that the CNI starts and this triggers the API service kicker to restart some services because a new interface appears. The immediate call to `enable dns` complains that it could not restart kubelet because another restart is on-going.

There is a way to guard against this by waiting for the calico daemon set to finish:
```
sudo microk8s.kubectl -n kube-system rollout status ds/calico-node
```

**Comment by ktsakalozos** [comment]
I do not think the change from `systemctl restart` to `snapctl restart` is the problem because the 1.20 uses `snapctl restart` and does not display this behavior.

**Comment by ktsakalozos** [comment]
It is also possible there is some lazy addition of network interfaces on the runners. Just added a `hostname -I` before the microk8s setup and I cannot reproduce the issue, maybe I was unlucky?

**Comment by ktsakalozos** [comment]
The problem is we detect the calico interface as a new interface that the API server needs to be configured for.

This is fixed
- on the master branch,
- on the 1.19 branch but has not landed on 1.19/stable yet, and
- I just created a PR to fix this on 1.20 (not sure why I have missed that).

I am sure you will not see this issue if you switch to 1.19/edge instead of 1.19/stable. Thank you for reporting this and appologies for the inconvenience this issue has caused.
